tests/gemd_query/test_gemd_query.py

```diff
@@ -0,0 +1,45 @@
+from uuid import uuid4
+import pytest
+
+from citrine.gemd_queries.criteria import PropertiesCriteria
+from citrine.gemd_queries.filter import AllRealFilter
+from citrine.gemd_queries.gemd_query import GemdQuery
+
+from tests.utils.factories import GemdQueryDataFactory
+
+
+def test_gemd_query_version():
+    valid = GemdQueryDataFactory()
```

Comment:
This uses the faker stuff below to make arbitrary GemdQueries?

Reply:
It doesn't do complex configurations, but it does fuzz values, yes.  

I started down the path of what it would take to generate different length lists, random branching, but there was more custom there than I was comfortable with.  Particularly given the future auto-generated question.

Reply:
I don't know what that would buy you at this level though?

Reply:
If it were easy, it buys you what fuzzing always buys you.  Instead, I just had to make sure the tested data structure was comprehensive.